lib/mayaUsd/render/MaterialXGenOgsXml/libraries/maya_utilities_ng.mtlx

```diff
@@ -0,0 +1,817 @@
+      <input name="value2" type="float" value="-1e+07" />
+      <input name="in1" type="vector3" nodename="ifUInBounds" />
+      <input name="in2" type="vector3" interfacename="defaultColor" />
+    </ifgreater>
+    <output name="outColor" type="vector3" nodename="ifVInBounds" />
+  </nodegraph>
+
+  <nodegraph name="MayaNG_fileTexture_vector4" nodedef="MayaND_fileTexture_vector4">
+    <!-- Note that for vector4 we do not consider the w channel to be an alpha -->
+    <invert name="invertedColor" type="vector4">
+      <input name="in" type="vector4" interfacename="inColor" />
+    </invert>
+    <ifequal name="invertIfRequired" type="vector4">
+      <input name="value1" type="boolean" interfacename="invert" />
+      <input name="value2" type="boolean" value="true" />
+      <input name="in1" type="vector4" nodename="invertedColor" />
+      <input name="in2" type="vector4" interfacename="inColor" />
+    </ifequal>
+    <power name="exposureGain" type="float">
+      <input name="in1" type="float" value="2.0" />
+      <input name="in2" type="float" interfacename="exposure" />
+    </power>
+    <multiply name="adjustedGain" type="vector4">
+      <input name="in1" type="vector4" interfacename="colorGain" />
+      <input name="in2" type="float" nodename="exposureGain" />
+    </multiply>
+    <multiply name="scaledColor" type="vector4">
+      <input name="in1" type="vector4" nodename="invertIfRequired" />
+      <input name="in2" type="vector4" nodename="adjustedGain" />
+    </multiply>
+    <add name="offsetColor" type="vector4">
+      <input name="in1" type="vector4" nodename="scaledColor" />
+      <input name="in2" type="vector4" interfacename="colorOffset" />
+    </add>
+    <swizzle name="inU" type="float">
+      <input name="in" type="vector2" interfacename="uvCoord" />
+      <input name="channels" type="string" uniform="true" value="x" />
+    </swizzle>
+    <ifgreater name="ifUInBounds" type="vector4">
+      <input name="value1" type="float" nodename="inU" />
+      <input name="value2" type="float" value="-1e+07" />
+      <input name="in1" type="vector4" nodename="offsetColor" />
+      <input name="in2" type="vector4" interfacename="defaultColor" />
+    </ifgreater>
+    <swizzle name="inV" type="float">
+      <input name="in" type="vector2" interfacename="uvCoord" />
+      <input name="channels" type="string" uniform="true" value="y" />
+    </swizzle>
+    <ifgreater name="ifVInBounds" type="vector4">
+      <input name="value1" type="float" nodename="inV" />
+      <input name="value2" type="float" value="-1e+07" />
+      <input name="in1" type="vector4" nodename="ifUInBounds" />
+      <input name="in2" type="vector4" interfacename="defaultColor" />
+    </ifgreater>
+    <output name="outColor" type="vector4" nodename="ifVInBounds" />
+  </nodegraph>
+
+  <!-- Nodegraph only wraps the difference in interface -->
+  <nodegraph name="MayaNG_clamp_vector3" nodedef="MayaND_clamp_vector3">
```

Comment:
New node. The Maya and MaterialX clamp nodes differ only in their interfaces. The NodeGraph allows using symmetric reader/writer code.